3-UsingAPIs/Week2/assignment/ex3-rollAnAce.js

```diff
@@ -12,21 +12,21 @@ Full description at: https://github.com/HackYourFuture/Assignments/blob/main/3-U
 // ! Do not change or remove the next two lines
 const rollDie = require('../../helpers/pokerDiceRoller');
 
-function rollDieUntil(wantedValue) {
-  // TODO: rewrite this function using async/await
-  return rollDie().then((value) => {
-    if (value !== wantedValue) {
-      return rollDieUntil(wantedValue);
-    }
-    return value;
-  });
+async function rollDieUntil(wantedValue) {
+  let value;
+  while ((value = await rollDie()) !== wantedValue) {
+    // Keep rolling until the value matches the wantedValue
+  }
```

Comment:
This works fine but it is not the most readable code. I would recommend keeping things simple, although maybe a bit less concise:

```suggestion
  let value = await rollDie();
  while ((value !== wantedValue) {
    value = await rollDie();
  }
```

You also no longer need the comment because now the code speaks for itself more clearly.